### PR TITLE
Product Page: Add QoL for screenreaders

### PR DESCRIPTION
**Pull request by jarstelfox**
Connects: https://github.com/iFixit/ifixit/issues/51161

### CODE REVIEW

File: packages/ui/cart/drawer/CartDrawer.tsx [hunk 1]
```diff
@@ -85,7 +86,12 @@ export function CartDrawer() {
                               {cart.isLoading ? (
                                  <Spinner size="xs" />
                               ) : (
-                                 cart.data?.totals.itemsCount ?? 0
+                                 <>
+                                    <VisuallyHidden>
+                                       Number of Items in Cart:
+                                    </VisuallyHidden>
+                                    {cart.data?.totals.itemsCount ?? 0}
```
Comment (ianrohde) [inline packages/ui/cart/drawer/CartDrawer.tsx:93]: Should this whole block be conditional based on `cart.data` being present?

Comment (jarstelfox) [reply packages/ui/cart/drawer/CartDrawer.tsx:93]: I think we intend to show `0` if there is no data

**Review by ianrohde — DISMISSED** [review]
CR ⚪

**Review by sterlinghirsh — APPROVED** [review]
CR :zap: Looks good to me as long as it doesn't majorly affect the appearance and fixes the screen reader issues.

**Comment by sterlinghirsh** [comment]
Looks like it's failing tests though.

**Review by sterlinghirsh — APPROVED** [review]
CR :zap:

**Comment by erinemay** [comment]
- Guide Section: Clarify for screen readers
   - I never experienced this on my screen reader.
- Next image: clarify slider buttons
   - Better and still usable
- Image Bullets: Give some context
   - Better and still usable 
- Cart: Clarify context
   - Now has "number of items in cart" before reading the number.
- Reviews section: Add clarity to review stats
   - It's kind of better in that there is more information on the page, but it's a little clunky. The progress bar stuff is unchanged from main. This is probably fine since it doesn't detract from the screen reader experience. There is more information

https://github.com/iFixit/react-commerce/assets/72166715/6dfd91ea-3475-4095-a2e6-5ed03ee89ce8

QA 🌵 I didn't notice any difference to the appearance of product pages or cart view.

**Pull request merged**